### ISSUE TITLE
Vulkan Attaching ViewportTexture to Material throws error

### ISSUE THREAD

**Issue by nonunknown** [godotengine/godot#50296]
### Godot version

v4.0.dev.custom_build [cb4e42155]

### System information

manjaro Gnome 21, AMD R7 200 series, Intel i3 3.0 GHZ,8GB RAM

### Issue description

When you select the SubViewport path to Viewport texture console starts spamming infinitely the error:

```
ERROR: Condition "!uniform_set" is true.
   at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:6697)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: draw_list_draw (drivers/vulkan/rendering_device_vulkan.cpp:6852)
```

After Closing godot the message continues:

```

ERROR: Condition "!material" is true.
   at: material_set_shader (servers/rendering/renderer_rd/renderer_storage_rd.cpp:1556)
ERROR: Attempting to use an uninitialized RID
   at: getornull (./core/templates/rid_owner.h:197)
ERROR: Condition "!material" is true.
   at: material_set_shader (servers/rendering/renderer_rd/renderer_storage_rd.cpp:1556)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:7853)
WARNING: 8 RIDs of type "UniformBuffer" were leaked.
     at: _free_rids (drivers/vulkan/rendering_device_vulkan.cpp:8378)

```

### Steps to reproduce

*Create Basic 3d Scene
* Create two cameras A and B
* Create a subviewport node
* attach camera B inside
* Create a meshinstance to apply a material
* attach a basic material to the mesh
* under albedo attach a new ViewportTexture
* Select the SubViewport
* Error should happen

### Minimal reproduction project

[ViewportTextureBug.zip](https://github.com/godotengine/godot/files/6787466/ViewportTextureBug.zip)

**Comment by GiantBlargg** [comment]
Is this fixed by #52528?

**Comment by akien-mga** [comment]
I tested and the error spam was fixed by #54489.

I'm not sure the ViewportTexture actually works though, there's still
```
ERROR: Viewport Texture must be set to use it.
   at: get_height (scene/main/viewport.cpp:113)
ERROR: Viewport Texture must be set to use it.
   at: get_width (scene/main/viewport.cpp:108)
```
and the preview is garbled, but that's probably something to track in a separate issue.
